packages/hop-node/src/chains/Chains/polygonzk/PolygonZkMessageService.ts

```diff
@@ -26,46 +26,41 @@ const polygonSdkVersion: Record<string, string> = {
   goerli: 'blueberry'
 }
 
-// TODO: Implement
-type MessageType = string
+type MessageOpts = {
+  messageDirection: MessageDirection
+}
+
+type Message = string
 type MessageStatus = string
 
-export class PolygonZkMessageService extends MessageService<MessageType, MessageStatus> implements IMessageService {
+export class PolygonZkMessageService extends AbstractMessageService<Message, MessageStatus, MessageOpts> implements IMessageService {
   ready: boolean = false
   apiUrl: string
   zkEvmClient: ZkEvmClient
 
-  constructor () {
-    super()
-
-    let l1Network: string | undefined
-    for (const network in networks) {
-      const chainId = networks[network as NetworkSlug]?.polygonzk?.networkId
-      if (chainId === this.chainId) {
-        l1Network = network
-        break
-      }
-    }
+  constructor (chainSlug: string) {
+    super(chainSlug)
 
-    if (!l1Network) {
-      throw new Error('polygon network name not found')
+    const networkSlug = getNetworkSlugByChainSlug(chainSlug)
+    if (!networkSlug) {
+      throw new Error(`Network slug not found for chain slug ${chainSlug}`)
     }
-
-    const polygonNetwork = polygonChainSlugs[l1Network]
+    const polygonNetwork: string = polygonChainSlugs[networkSlug]
     this.apiUrl = `https://proof-generator.polygon.technology/api/v1/${polygonNetwork}/block-included`
 
     use(Web3ClientPlugin)
     setProofApi('https://proof-generator.polygon.technology/')
 
     this.zkEvmClient = new ZkEvmClient()
 
-    this.#init(l1Network)
+    this.#init(networkSlug)
       .then(() => {
         this.ready = true
         this.logger.debug('zkEVM client initialized')
       })
       .catch((err: any) => {
         this.logger.error('zkEvmClient initialize error:', err)
+        throw err
```

Comment:
I think the `throw` here will result in an uncaught promise error in nodejs, so maybe exiting with `process.exit(1)` would be better

Some recommendations from chatgpt

- Event Emitter: Use an event emitter to emit an error event from the constructor and handle it outside the class.
- Initialization Method: Instead of initializing in the constructor, provide an explicit initialization method that returns a Promise. This allows the calling code to handle initialization asynchronously and catch any errors.
- Factory Function: Create a static factory method that handles the asynchronous initialization and returns a Promise of the instance.

Reply:
Good call, updated to `process.exit(1)`